R/plot-build.R

```diff
@@ -481,3 +398,132 @@ table_add_tag <- function(table, label, theme) {
     t = place$t, l = place$l, b = place$b, r = place$r
   )
 }
+
+# Add the legends to the gtable
+table_add_legends <- function(table, legends, theme) {
+
+  # Extract sizes
+  widths <- heights <- set_names(
+    rep(list(unit(0, "cm")), length(legends)),
+    names(legends)
+  )
+  if (is.zero(legends)) {
+    legends <- rep(list(zeroGrob()), 5)
+    names(legends) <- c(.trbl, "inside")
+  }
+  empty <- vapply(legends, is.zero, logical(1))
+  widths[!empty]  <- lapply(legends[!empty], gtable_width)
+  heights[!empty] <- lapply(legends[!empty], gtable_height)
+  spacing <- theme$legend.box.spacing %||% unit(0.2, "cm")
+
+  if (!is.zero(legends$inside)) {
+    # Set the justification of the legend box
+    just <- valid.just(calc_element("legend.justification.inside", theme))
+
+    position <- theme$legend.position.inside %||% just
+    legends$inside <- editGrob(
+      legends$inside,
+      vp = viewport(
+        x = position[1],
+        y = position[2],
+        just = just,
+        height = heights$inside,
+        width  = widths$inside
+      )
+    )
+  }
+
+  legends[.trbl] <- lapply(.trbl, function(position) {
+    box <- .subset2(legends, position)
```

Comment:
Is `legends` a weird class that doesn't respond correctly to `[[`-indexing? If not we should use that